### ISSUE TITLE
Early stop to prevent overfitting

### ISSUE THREAD

**Issue by fdfrontdev** [dhopp1/nowcast_lstm#9]
Hi,

would like to suggest adding support for early stopping and choose best weight based on lowest loss or highest accuracy, or highest validation accuracy.

**Comment by dhopp1** [comment]
Hello, this can more or less be done via the `train_episodes` parameter. A lower number will stop the training earlier.

If you mean choosing hyperparameters via lowest loss, you can do that with this function: `from nowcast_lstm.model_selection import hyperparameter_tuning`.

If you're referring to dynamically halting training based on a certain in-sample loss threshold, this is indeed on my list to add in the future, but I'm not sure when I will get around to adding it. In the interim you can tune the `train_episodes` parameter to achieve a similar result.
